src/Components/RightClick/Base/RightClick.tsx

```diff
@@ -0,0 +1,153 @@
+// Libraries
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  RefObject,
+  MouseEvent,
+  forwardRef,
+} from 'react'
+import {createPortal} from 'react-dom'
+import uuid from 'uuid'
+
+// Components
+import {RightClickMenu, RightClickMenuRef} from './RightClickMenu'
+
+// Utils
+import {createPortalElement, destroyPortalElement} from '../../../Utils/index'
+
+// Styles
+import './RightClick.scss'
+
+// Types
+import {
+  ComponentColor,
+  StandardFunctionProps,
+  Coordinates,
+} from '../../../Types'
+
+export interface RightClickProps extends StandardFunctionProps {
+  /** RightClick menu color */
+  color?: ComponentColor
+  /** Callback function fired when state changes to "show" */
+  onShow?: () => void
+  /** Callback function fired when state changes to "hide" */
+  onHide?: () => void
+  /** Prevents the right click menu from firing */
+  disabled?: boolean
+  /** Reference to trigger element */
+  triggerRef: RefObject<any>
+}
+
+export type RightClickRef = RightClickMenuRef
+
+const rightClickPortalName = 'right-click'
+
+export const RightClickRoot = forwardRef<RightClickRef, RightClickProps>(
+  (
+    {
+      id,
+      style,
+      onShow,
+      onHide,
+      children,
+      className,
+      triggerRef,
+      disabled = false,
+      testID = 'right-click',
+      color = ComponentColor.Default,
+    },
+    ref
+  ) => {
+    const [expanded, setExpanded] = useState<boolean>(false)
+    const [portalID, setPortalID] = useState<string>('')
+    const mouseOffset = useRef<Coordinates>({x: 0, y: 0})
+
+    useEffect((): (() => void) => {
+      const newPortalID = `cf-${rightClickPortalName}-portal-${uuid.v4()}`
+      setPortalID(newPortalID)
+
+      createPortalElement(newPortalID, rightClickPortalName)
+      handleAddEventListeners()
+
+      return (): void => {
```

Comment:
nit: you should normally reverse the order of your destructors to defensively manage side effects that might sneak in, but not a big deal.